src/libos/crates/vdso-time/src/lib.rs

```diff
@@ -9,289 +7,284 @@ extern crate sgx_types;
+            vdso_ocall_get_vdso_info(
+                &mut ret as *mut _,
+                &mut vdso_addr as *mut _,
+                release.as_mut_ptr(),
+                release.len() as _,
+            );
+        }
+        if ret != 0 {
+            return_errno!(EINVAL, "Vdso vdso_ocall_get_vdso_info() failed")
+        }
 
-        #[cfg(feature = "sgx")]
-        let (vdso_addr, hres_resolution, coarse_resolution, tss, release) = {
-            extern "C" {
-                fn vdso_ocall_get_vdso_info(
-                    ret: *mut libc::c_int,
-                    vdso_addr: *mut libc::c_ulong,
-                    hres_resolution: *mut libc::c_long,
-                    coarse_resolution: *mut libc::c_long,
-                    release: *mut libc::c_char,
-                    release_len: libc::c_int,
-                    tss: *mut timespec,
-                    tss_len: libc::c_int,
-                ) -> sgx_types::sgx_status_t;
-            }
+        Self::match_kernel_version(vdso_addr, &release)
+    }
 
-            let mut vdso_addr: libc::c_ulong = 0;
-            let mut hres_resolution: libc::c_long = 0;
-            let mut coarse_resolution: libc::c_long = 0;
-            let mut release = [0 as libc::c_char; 65];
-            let mut tss = [timespec {
-                tv_sec: 0,
-                tv_nsec: 0,
-            }; VDSO_BASES as usize];
-            let mut ret: libc::c_int = 0;
-            unsafe {
-                vdso_ocall_get_vdso_info(
-                    &mut ret as *mut _,
-                    &mut vdso_addr as *mut _,
-                    &mut hres_resolution as *mut _,
-                    &mut coarse_resolution as *mut _,
-                    release.as_mut_ptr(),
-                    release.len() as _,
-                    tss.as_mut_ptr(),
-                    tss.len() as _,
-                );
-            }
-            if ret != 0 {
-                return Err(());
-            }
+    #[cfg(not(feature = "sgx"))]
+    fn get_vdso_data_ptr_from_host() -> Result<VdsoDataPtr> {
+        const AT_SYSINFO_EHDR: u64 = 33;
+        let vdso_addr = unsafe { libc::getauxval(AT_SYSINFO_EHDR) };
 
-            let hres_resolution = if hres_resolution != 0 {
-                Some(hres_resolution)
-            } else {
-                None
-            };
-            let coarse_resolution = if coarse_resolution != 0 {
-                Some(coarse_resolution)
-            } else {
-                None
-            };
+        let mut utsname: libc::utsname = unsafe { std::mem::zeroed() };
+        let ret = unsafe { libc::uname(&mut utsname as *mut _) };
+        if ret != 0 {
+            return_errno!(EINVAL, "Vdso get utsname failed");
+        }
+        let release = utsname.release;
 
-            (vdso_addr, hres_resolution, coarse_resolution, tss, release)
-        };
+        Self::match_kernel_version(vdso_addr, &release)
+    }
 
+    fn get_checked_vdso_addr(vdso_addr: u64) -> Result<u64> {
```

Comment:
Similar to the other comment, we don't want call a method `get_xx` if it is not a getter actually. Renaming this method to `check_vdso_addr` seems to be a good idea to me. And you don't need to consume the subject of a check method (like you do for `get_checked_vdso_addr` or `get_checked_vdso`). A signature of `fn check_vdso_addr(vdso_addr: &u64) -> Result<()>` looks fine to me.

Reply:
changed to `fn check_vdso_addr(vdso_addr: u64) -> Result<()>`